### PR TITLE
Trafficgen trigger config

### PR DESCRIPTION
**Pull request by atheurer**
Updates tool-trigger to allow sample number
Updates pbench-trafficgen to use tool triggers and specify the benchmark period that tools are used

### CODE REVIEW

File: agent/bench-scripts/pbench-trafficgen [hunk 1]
```diff
@@ -88,6 +88,7 @@ packet_protocol="UDP"
 pre_sample_cmd=""
 traffic_profiles=""
 sysinfo="none"
+tool_period="binary-search" # can also be "repeat-final-vallidation"
```
Comment (portante) [inline agent/bench-scripts/pbench-trafficgen]: s/vallidation/validation/

File: agent/bench-scripts/pbench-trafficgen [hunk 2]
```diff
@@ -670,6 +683,16 @@ if [ $one_shot == "y" ]; then
 	cmd="$cmd --one-shot=1"
 fi
 
+# register the tool triggers depending on the tool period preference
+if [ "$tool_period" == "binary-search" ]; then
+	pbench-register-tool-trigger --group=$tool_group --start-trigger="Starting binary-search" --stop-trigger="Finished binary-search"
+else
```
Comment (portante) [inline agent/bench-scripts/pbench-trafficgen]: Can you consider using `elif` instead to avoid a level of indentation?

**Comment by portante** [comment]
Also, can you rebase these changes to remove the conflicts?

File: agent/bench-scripts/pbench-trafficgen (hunk 1 above)
Comment (atheurer) [reply agent/bench-scripts/pbench-trafficgen]: taken care of with new commit

File: agent/bench-scripts/pbench-trafficgen (hunk 2 above)
Comment (atheurer) [reply agent/bench-scripts/pbench-trafficgen]: sure

**Comment by atheurer** [comment]
> Also, can you rebase these changes to remove the conflicts?

@portante are you seeing conflicts with base branch?  Github reports no conflicts when I view this PR

**Comment by portante** [comment]
@atheurer, you patch appears to contain 6 commits, and github is reporting that the branch cannot be rebased due to conflicts.

If you do, `git remote update`, and then `git rebase` on your branch, it might clear things up.

**Comment by atheurer** [comment]
> @atheurer, you patch appears to contain 6 commits, and github is reporting that the branch cannot be rebased due to conflicts.

I have no idea why, but 4 of those commits are from you, I am assuming from unrelated work.
 
> If you do, `git remote update`, and then `git rebase` on your branch, it might clear things up.

I did this before doing the PR, but did it again just now, no change:

[atheurer@perf84 pbench]$ git remote update
Fetching origin
Fetching upstream
[atheurer@perf84 pbench]$ git rebase
Current branch trafficgen-trigger-config is up to date.

Here is my 'git remote -v':
[atheurer@perf84 pbench]$ git remote -v
origin	git@github.com:atheurer/pbench.git (fetch)
origin	git@github.com:atheurer/pbench.git (push)
upstream	git@github.com:distributed-system-analysis/pbench.git (fetch)
upstream	git@github.com:distributed-system-analysis/pbench.git (push)

**Comment by portante** [comment]
Ah, try `git branch --set-upstream-to=upstream/master`, then `git rebase`.

**Pull request merged**